### PR TITLE
Add URL normalise to image optimiser URL

### PR DESCRIPTION
**Pull request by mathvbarone**
If Basepath is set, it needs to be removed from the query

 Not normalised -> error 403
 `url: '<base-path>/assets/images/logo.svg'`
 
Normalised -> 200
url: `'/assets/images/logo.svg'`

### CODE REVIEW

File: packages/libs/core/src/images/imageOptimizer.ts [hunk 1]
```diff
@@ -115,6 +116,20 @@ export async function imageOptimizer(
...
+   * /<base-path>/_next/image?url=%2F<base-path>%2Fassets%2Fimages%2Flogo.svg&w=256&q=75
+   * 
+   * Normalised -> 200
+   * /<base-path>/_next/image?url=%2Fassets%2Fimages%2Flogo.svg&w=256&q=75
+   * 
+   * 
+   */
+  if (url.startsWith(basePath)) {
+    url = url.slice(basePath.length);
+  }
```
Comment (ivanbabenko) [inline packages/libs/core/src/images/imageOptimizer.ts:132]: Maybe add blank line after this condition for readability

Comment (mathvbarone) [reply packages/libs/core/src/images/imageOptimizer.ts:132]: Done! 😛

**Comment by dphang** [comment]
Thanks, though it looks like the actual Next.js behavior with local `next start` is that basepath should be supplied to the URL when it's set in config, otherwise if no basepath there is a 400 response returned. So I guess in another PR you should actually check if basepath is passed into the relative URLs as well and throw a 400 response if not passed (to be fully in parity with Next.js behavior).

Not sure how it is related to #1620? I think that one is a path issue in uploading assets via the CDK construct.

**Comment by mathvbarone** [comment]
> Thanks, though it looks like the actual Next.js behavior with local `next start` is that basepath should be supplied to the URL when it's set in config, otherwise if no basepath there is a 400 response returned. So I guess in another PR you should actually check if basepath is passed into the relative URLs as well and throw a 400 response if not passed (to be fully in parity with Next.js behavior).
> 
> Not sure how it is related to #1620? I think that one is a path issue in uploading assets via the CDK construct.

Hey, @dphang I added the verification as requested, can you please take another look?

I thought #1620 was also related to the addition of basepath on the query, you can disregard that (I'll edit the description of the PR)

Thanks!

File: packages/libs/core/src/images/imageOptimizer.ts [hunk 2]
```diff
@@ -116,7 +133,14 @@ export async function imageOptimizer(
   let isAbsolute: boolean;
 
   if (url.startsWith("/")) {
-    href = url;
+    // Ensure that Basepath is in the URL, otherwise, a 400 is triggered (same behaviour as Nextjs)
+    if (basePath !== "/" && !url.startsWith(basePath)) {
```
Comment (dphang) [inline packages/libs/core/src/images/imageOptimizer.ts:137]: Ideally should add tests for basepath (right now) but you can add in another PR

**Review by dphang — APPROVED** [review]

**Comment by dphang** [comment]
e2e tests passed: https://github.com/serverless-nextjs/serverless-next.js/runs/4145408350?check_suite_focus=true

**Pull request merged**